src/app/health/health-event.component.ts

```diff
@@ -46,6 +46,20 @@ export class HealthEventComponent implements OnInit {
     });
   }
 
+  isEmptyForm() {
+    let notAllEmpty: Boolean;
+    Object.keys(this.healthForm.controls).forEach((key: string) => {
```

Comment:
Two things you can use here to improve:

1. Use [Object.values()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values) to access the values instead of the keys
2. Use [Array.some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) to return a true/false in one line

Reply:
@paulbert I found the value from Object.values() is not the value I want. It still a AbstractControl Object. Let say the element from Object.values() is `v`, The value used to make a verification should be the `v.value`.